garak/generators/nim.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python3
+
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""NVIDIA Inference Microservice LLM interface"""
+
+import json
+import logging
+import os
+from typing import List, Union
+
+import requests
+
+import backoff
+
+from garak import _config
+from garak.generators.base import Generator
```

Comment:
I'm inclined to offer two separate classes, one for chat and one for completions, and default to chat. maintaining model names in here builds in technical debt and unfortunate UX. what do you think?

Reply:
I agreed maintaining the names would be great to avoid, I can see a direct selections as reasonable such as:
OpenAIChat be default:
```
python -m garak -m nim -n <chat_model> -g 1 -p dan.DanInTheWild
python -m garak -m nim.OpenAIChat -n <chat_model> -g 1 -p dan.DanInTheWild
```
And for completion required a specific class:
```
python -m garak -m nim..OpenAICompletion -n <completion_model> -g 1 -p dan.DanInTheWild
```

This then lends to future support for NemoLLM NIMs as:
```
python -m garak -m nim.NemoLLMChat -n <chat_model> -g 1 -p dan.DanInTheWild
python -m garak -m nim.NemoLLMCompletion -n <completion_model> -g 1 -p dan.DanInTheWild
```

Reply:
yes! agree. excellent. i'll do that.